pkg/storage/etcd/etcd_helper.go

```diff
@@ -191,6 +191,23 @@ func (h *etcdHelper) Set(ctx context.Context, key string, obj, out runtime.Objec
 	if ctx == nil {
 		glog.Errorf("Context is nil")
 	}
+
+	version := uint64(0)
+	if h.versioner != nil {
+		var err error
+		if version, err = h.versioner.ObjectResourceVersion(obj); err != nil {
+			return errors.New("couldn't get resourceVersion from object")
+		}
+		if version != 0 {
+			// We cannot store object with resourceVersion in etcd, we need to clear it here.
+			if err := h.versioner.UpdateObject(obj, nil, 0); err != nil {
+				return errors.New("resourceVersion cannot be set on objects store in etcd")
+			}
+		}
+	} else {
+		return errors.New("couldn't access version without versioner")
```

Comment:
Hrm - isn't this a change in behavior?  We're failing now if you don't set a versioner, whereas before we just continued.  I'm not sure the check should be at this level - if we require versioner, we should fail earlier.  Would suggest not adding the else condition for now.